dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/datadeposit/UrlManagerServiceBean.java

```diff
@@ -172,10 +176,8 @@ public String getHostnamePlusBaseUrlPath(String url) throws SwordError {
             optionalPort = ":" + port;
         }
         String requestedHostname = u.getHost();
-        String hostName = systemConfig.getFqdn();
-        if (hostName == null) {
-            hostName = "localhost";
-        }
+        String hostName = systemConfig.getDataverseServer();
```

Comment:
You can just use SiteUrl, getting the url, stripping it, then creating it again below is redundant.

Reply:
Not entirely true. `systemConfig.getDataverseServer()` returns only hostname without port, which is being taken from argument passed to the method.

Reply:
We've agreed on simplifying this method to just return `systemConfig.getDataverseSiteUrl() + SwordConfigurationConstants.BASE_URL_PATH_CURRENT`